applications/MappingApplication/README.md

```diff
@@ -0,0 +1,182 @@
+
+~~~py
+mapper.Map(KM.REACTION, KM.FORCE, KratosMapping.Mapper.USE_TRANSPOSE)
+~~~
+
+#### Updating the Interface
+In case of moving interfaces (e.g. in a problem involving Contact between bodies) it can become necessary to update the _Mapper_ to take the new geometrical positions of the interfaces into account.\
+One way of doing this would be to construct a new _Mapper_, but this is not efficient and sometimes not even possible.
+
+Hence the _Mapper_ provides the **UpdateInterface** function for updating itseld with respect to the new geometrical positions of the interfaces.\
+Note that this is potentially an expensive operation due to searching the new geometrical neighbors on the interface.
+
+~~~py
+mapper.UpdateInterface()
+~~~
+
+#### Checking which mappers are available
+The following can be used to see which _Mappers_ are available:
+~~~py
+# available mappers for shared memory
+KratosMapping.MapperFactory.GetRegisteredMapperNames()
+
+# available mappers for distributed memory
+KratosMapping.MapperFactory.GetRegisteredMPIMapperNames()
+
+# check if mapper for shared memory exists
+KratosMapping.MapperFactory.HasMapper("mapper_name")
+
+# check if mapper for distributed memory exists
+KratosMapping.MapperFactory.HasMPIMapper("mapper_name")
+~~~
+
+
+### Available Mappers
+This section explains the theory behind the mappers.
+
+#### Nearest Neighbor
+The _NearestNeighborMapper_ is a very simple/basic _Mapper_. Searches its closest neighbor (node) on the other interface. During mapping it gets/sets its value to the value of its closest neighbor.
+
+This mapper is best suited for problems where both interfaces have a similar discretization. Furthermore it is very robust and can be used for setting up problems when one does not (yet) want to deal with mapping.
+
+Internally it constructs the mapping matrix, hence it offers the usage of the transposed mapping matrix. When using this, for very inhomogenous interface discretizations it can come to oscillations in the mapped quantities.
+
+#### Nearest Element
+The _NearestElementMapper_ projects nodes to the elements( or conditions) on other side of the inteface. Mapping is then done by interpolating the values of the nodes of the elements by using the shape functions at the projected position.
+
+This mapper is best suited for problems where the _NearestNeighborMapper_ cannot be used, i.e. for cases where the discretization on the interfaces is different. Note that it is less robust than the _NearestNeighborMapper_ due to the projections it performs. In case a projection fails it uses an approximation that is similar to the approach of the _NearestNeighborMapper_.
+
+Internally it constructs the mapping matrix, hence it offers the usage of the transposed mapping matrix. When using this, for very inhomogenous interface discretizations it can come to oscillations in the mapped quantities.
```

Comment:
Is not very clear if the oscillation happens when using transpose or not.

Reply:
clarified

Reply:
clarified